### ISSUE TITLE
read env before starting an app

### ISSUE THREAD

**Issue by diversys** [humdingerb/quicklaunch#15]
qBittorrent doesn't seem to start because of it.
Also https://github.com/threedeyes/QtHaikuStyle is ignored because of that as it requires

`export QT_STYLE_OVERRIDE=haiku` 
from /boot/home/config/settings/boot/UserSetupEnvironment

**Comment by diversys** [comment]
This was also a reason behind Pe crash here https://github.com/HaikuArchives/Pe/issues/29

**Comment by diversys** [comment]
Could probably be used http://cgit.haiku-os.org/haiku/tree/src/servers/launch/BaseJob.cpp#n196

**Comment by diversys** [comment]
Seems to be a dupe https://dev.haiku-os.org/ticket/12534. Would be nice to workaround it with something like` putenv("HOME=/boot/home")` for the time being as some apps don't even lunch or crash. What do you think?

**Comment by humdingerb** [comment]
You mean simply putting "putenv("HOME=/boot/home");" into QL's main()?
Do you have another example of an app not starting besides qBittorrent? Can't install that ATM in gcc2h.

**Comment by diversys** [comment]
See this commit https://github.com/threedeyes/qthaikuplugins/commit/df1dc07 on how to better workaround it. IIRC, it was Marble, however it does start now for some reason.

**Comment by miqlas** [comment]
fish-shell also does strange things with QL.
I suppose XDG vars doesn't exported. IDK why.

**Comment by phw** [comment]
I digged a bit. The reason why Picard is not running is missing `LIBRARY_PATH` environment variable, which break Python ctypes searching for library. Could work around that in some way.

A native app behaving different is Terminal. Normally when you start Terminal the shell opens in the home directory. If started via Quicklaunch it starts at /.

I think QuickLaunch should launch apps exactly like it happens if you start it from the Deskbar. Apps should end up with the same environment in both cases.

**Comment by humdingerb** [comment]
Agreed. But I suppose that's something for the OS to do right, not for apps to work around (which I still don't know how anyway). There's a ticket on Haiku's Trac, as mentioned above.

**Comment by phw** [comment]
For reference this seems to be the ticket: https://dev.haiku-os.org/ticket/12534

And the dependency of Python on `LD_LIBRARY_PATH` is actually part of the Haiku patchset at https://github.com/haikuports/haikuports/blob/master/dev-lang/python/patches/python3-3.8.8.patchset#L655

**Comment by leavengood** [comment]
I think I've gotten to the root of this issue, though we are unsure of the proper fix, but it will probably involve fixing how Shortcuts launches apps. More details in this forum thread: https://discuss.haiku-os.org/t/environment-variables-behavior-on-haiku/10454/24

One way you guys could verify my theory is to run QuickLaunch from Tracker or Deskbar and see if some of the environment variables show up then. Though the only ones you will see in that case is those defined in [SetupEnvironment](https://github.com/haiku/haiku/blob/master/data/system/boot/SetupEnvironment), not ones from the bash profile (I am fairly sure.) It might be easier to use a test app which puts the environment contents in a BTextView though I know that might not be possible for some people. When I get a chance I plan to write that app to sanity check any fixes. 😄
